src/queue/SoundQueue.ts

```diff
@@ -75,10 +75,14 @@ export default class SoundQueue {
       .catch(error => console.error('Error occured!', '\n', error));
   }
 
-  private deafen(connection: VoiceConnection) {
+  private deafen(connection: VoiceConnection): Promise<VoiceConnection> {
```

Comment:
The type is incorrect, because you decided to reject with a `string`. Is the type annotation necessary? I'm sure it is correctly inferred.

I think you can just resolve with `connection` here. This is not that important.